rpc/dcrd.go

```diff
@@ -261,3 +264,51 @@ func (c *DcrdRPC) ExistsLiveTicket(ticketHash string) (bool, error) {
 
 	return bitset.Bytes(existsBytes).Get(0), nil
 }
+
+func (c *DcrdRPC) GetBlockVerbose(hash string) (*dcrdtypes.GetBlockVerboseResult, error) {
+	var resp dcrdtypes.GetBlockVerboseResult
+	const verbose = true
+	const verboseTx = true
+	err := c.Call(context.TODO(), "getblock", &resp, hash, verbose, verboseTx)
+	if err != nil {
+		return nil, err
+	}
+	return &resp, nil
+}
+
+func (c *DcrdRPC) GetBlockHash(height int64) (string, error) {
+	var resp string
+	err := c.Call(context.TODO(), "getblockhash", &resp, height)
+	if err != nil {
+		return "", err
+	}
+	return resp, nil
+}
+
+// GetCFilterV2 retrieves the GCS filter for the provided block hash and returns
+// it along with the key derived from the provided merkle root.
+func (c *DcrdRPC) GetCFilterV2(blockHash string, merkleRoot string) ([gcs.KeySize]byte, *gcs.FilterV2, error) {
+	var key [gcs.KeySize]byte
+	var resp dcrdtypes.GetCFilterV2Result
+	err := c.Call(context.TODO(), "getcfilterv2", &resp, blockHash)
+	if err != nil {
+		return key, nil, fmt.Errorf("getcfilterv2 error: %w", err)
+	}
+
+	filterB, err := hex.DecodeString(resp.Data)
+	if err != nil {
+		return key, nil, fmt.Errorf("error decoding block filter: %w", err)
+	}
+
+	filter, err := gcs.FromBytesV2(blockcf2.B, blockcf2.M, filterB)
```

Comment:
Since this is happening via RPC and the VSP probably controls the dcrd that is providing the filters, it isn't a huge concern, but this should ideally check the inclusion proof as well once the header commitments are active (e.g. block `000000000000000010815bed2c4dc431c34a859f4fc70774223dde788e95a01e` (height 431488) on mainnet).

e.g.
```Go
func (c *DcrdRPC) GetCFilterV2(blockHash string, header *wire.BlockHeader, verifyProof bool) ([gcs.KeySize]byte, *gcs.FilterV2, error) {
	...

	if verifyProof {
		filterHash := filter.Hash()
		if !standalone.VerifyInclusionProof(&header.StakeRoot, &filterHash, resp.ProofIndex, resp.ProofHashes) {
			/* error */
		}
	}

	return blockcf2.Key(&header.MerkleRoot), filter, nil
}
```